packages/next/src/utils.ts

```diff
@@ -1215,10 +1215,14 @@ let _usesSrcCache: boolean | undefined;
 
 async function usesSrcDirectory(workPath: string): Promise<boolean> {
   if (!_usesSrcCache) {
-    const source = path.join(workPath, 'src', 'pages');
+    const sourcePages = path.join(workPath, 'src', 'pages');
+    const sourceAppdir = path.join(workPath, 'src', 'app');
 
     try {
-      if ((await fs.stat(source)).isDirectory()) {
+      if (
+        (await fs.stat(sourcePages)).isDirectory() ||
+        (await fs.stat(sourceAppdir)).isDirectory()
```

Comment:
You'll need two different try/catch because usesSrcDirectory() would need to return true if either one exists. Right now, this would throw when `sourcePages` doesn't exist but  `sourceAppdir` does exist.

Reply:
ah, thanks! fixed it